File: python/math/171. Excel Sheet Column Number.py
'''
own
Time: O(n)
Space: O(1)
'''

class Solution:
    # own
    def titleToNumber(self, columnTitle: str) -> int:
        res = 0

        for i in range(len(columnTitle)):
            digit = len(columnTitle) - 1 - i  # 0 ~ len(columnTitle) 
            val = ord(columnTitle[i]) - 64  # 1 ~ 26

            tempRes = 26 ** digit * val if digit > 0 else val

            res += tempRes

        return res

File: javascript/math/171. Excel Sheet Column Number.js
/**
 * @param {string} columnTitle
 * @return {number}
 */
var titleToNumber = function(columnTitle) {
    let res = 0;
    let tempRes = 0;
    for (let i=0; i<columnTitle.length; i++) {
        let digit = columnTitle.length -1 - i;
        let val = columnTitle[i].charCodeAt(0) - 64;

        digit > 0 ? tempRes = 26 ** digit * val : tempRes = val;

        res += tempRes;
    }
    return res;
};